### PR TITLE
Fix config reload after vim edit

### PR DESCRIPTION
**Pull request by chabou**
When `vim` edit a file, it write a temp file and move it.
In this case, our watcher didn't work anymore.

This PR rewatch file when file is renamed/mv

### CODE REVIEW

File: app/config.js [hunk 1]
```diff
@@ -35,8 +36,18 @@ const _watch = function() {
...
+    return;
+  }
+  // macOS/Linux
+  setWatcher();
+  function setWatcher() {
+    _watcher = fs.watch(cfgPath, eventType => {
+      if (eventType === 'rename') {
+        _watcher.close();
+        // Ensure that new file has been written
+        setTimeout(() => setWatcher(), 500);
```
[the remark this answers is not in the document]
Comment (chabou) [reply app/config.js]: What would be our retry strategy? Only one time? Retry until it works?
If we think that 500ms is not enough, we need to put some try/catch too.

But I really think that it would work 99.99%. If not, maybe we'll add some retries later?

Mmmmh I should try this with config file remove...

Do not merge it please, I think it needs some more tests on some edge cases

**Comment by chabou** [comment]
I made a lot of test and it works really great.
I added a try/catch but it should be unused. If a config file is missing, a new one is written: https://github.com/zeit/hyper/blob/24b83615d4f37cfead075c5a80383dac9c3311ee/app/config/import.js#L41

If writing failed, an exception will be shown.
Thus, we can silently (but with log) discard an hypothetical watch exception.
User would have to restart its application to have a config watcher, but this is ok for me because it should not happen.
